### PR TITLE
docs(compute-samples): improving snapshot samples

### PR DESCRIPTION
**Pull request by Sita04**
Improving the code snippets for disk snapshots, so they can be used on this page: [Create and manage disk snapshots](https://cloud.google.com/compute/docs/disks/create-snapshots)

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/disks/DeleteSnapshotsByFilter.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+  // Deletes all snapshots in project that meet the filter criteria.
+  public static void deleteSnapshotsByFilter(String projectId, String filter)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+
+    for (Snapshot snapshot : listSnapshots(projectId, filter).iterateAll()) {
+      deleteSnapshot(projectId, snapshot.getName());
+    }
+  }
```
Comment (kurtisvg) [inline compute/cloud-client/src/main/java/compute/disks/DeleteSnapshotsByFilter.java:89]: I don't think these samples should be written as 3 separate functions - it makes it harder to read and counteracts the advice we give to reuse the client.

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: compute/cloud-client/src/main/java/compute/disks/GetSnapshot.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  // Get information about a snapshot.
+  public static void getSnapshot(String projectId, String snapshotName) throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `snapshotsClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (SnapshotsClient snapshotsClient = SnapshotsClient.create()) {
+
```
[the remark this answers is not in the document]
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/disks/GetSnapshot.java:43]: Done.

File: compute/cloud-client/src/main/java/compute/disks/DeleteSnapshotsByFilter.java (hunk 1 above)
Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/disks/DeleteSnapshotsByFilter.java:89]: I replicated the method structure from what we have in python (for consistency). https://github.com/googleapis/python-compute/pull/276

Will talk to the team for reformatting this sample. Thanks.

Comment (Sita04) [reply compute/cloud-client/src/main/java/compute/disks/DeleteSnapshotsByFilter.java]: Thanks Maciej.

@kurtisvg I've updated the code sample.

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**